matrix-sdk/src/main/java/org/matrix/androidsdk/db/MXMediaDownloadWorkerTask.java

```diff
@@ -905,15 +941,61 @@ public void run() {
 
         // build a JSON from the error
         if (!TextUtils.isEmpty(defaultError.error)) {
-            mErrorAsJsonElement = JsonUtils.getGson(false).toJsonTree(defaultError);
+            jsonElementResult = JsonUtils.getGson(false).toJsonTree(defaultError);
         }
 
-        // remove the image from the loading one
+        // remove the task from the loading one
         synchronized (mPendingDownloadById) {
             mPendingDownloadById.remove(mDownloadId);
         }
 
-        return null;
+        return jsonElementResult;
+    }
+
+    /**
+     * Get the public key of the antivirus server
+     *
+     * @return either empty string if server does not provide the public key, null in case of error, or the public server key
+     */
+    private String getAntivirusServerPublicKey() {
+        // Make async request sync with a CountDownLatch
+        final CountDownLatch latch = new CountDownLatch(1);
+        final String[] publicServerKey = new String[1];
+
+        if (mMediaScanRestClient == null) {
+            return "";
```

Comment:
I would prefer to return null for the moment in this case in order to block the download

Reply:
it will impact Riot and any other client using the SDK, I think it is not a good idea...

Reply:
ok